tests/test_bar.py

```diff
@@ -1,70 +1,192 @@
+
+    def u(mu, sigma, x):
+        return (x - mu) ** 2 / (2 * sigma ** 2)
+
+    mu_a, sigma_a = params_a
+    mu_b, sigma_b = params_b
+
+    u_a = partial(u, mu_a, sigma_a)
+    u_b = partial(u, mu_b, sigma_b)
+
+    rng = np.random.default_rng(seed)
+
+    x_a = rng.normal(mu_a, sigma_a, (n_samples,))
+    x_b = rng.normal(mu_b, sigma_b, (n_samples,))
+
+    u_kln = np.array([[u_a(x_a), u_a(x_b)], [u_b(x_a), u_b(x_b)]])
+
+    dlogZ = np.log(sigma_a) - np.log(sigma_b)
+
+    return u_kln, dlogZ
+
+
+def make_partial_overlap_uniform_ukln_example(dlogZ: float, n_samples: int = 100) -> NDArray:
+    """Generate 2-state u_kln matrix for uniform distributions with partial overlap"""
+
+    def u_a(x):
+        """Unif[0.0, 1.0], with log(Z) = 0"""
+        in_bounds = (x > 0) * (x < 1)
+        return np.where(in_bounds, 0, +np.inf)
+
+    def u_b(x):
+        """Unif[0.5, 1.5], with log(Z) = dlogZ"""
+        x_ = x - 0.5
+        return u_a(x_) + dlogZ
+
+    rng = np.random.default_rng(2023)
+
+    x_a = rng.uniform(0, 1, (n_samples,))
+    x_b = rng.uniform(0.5, 1.5, (n_samples,))
+
+    assert np.isfinite(u_a(x_a)).all()
+    assert np.isfinite(u_b(x_b)).all()
+
+    u_kln = np.array([[u_a(x_a), u_a(x_b)], [u_b(x_a), u_b(x_b)]])
 
-@pytest.mark.nogpu
-def test_bootstrap_bar():
+    return u_kln
+
+
+@pytest.mark.parametrize("sigma", [0.1, 1.0, 10.0])
+def test_bootstrap_bar(sigma):
     np.random.seed(0)
-    n_bootstrap = 1000
+    n_bootstrap = 100
 
-    for sigma_F in [0.1, 1, 10]:
-        # default rbfe instance size, varying difficulty
-        w_F, w_R = gaussian_work_example(2000, 2000, sigma_F=sigma_F, seed=0)
+    # default rbfe instance size, varying difficulty
+    u_kln, dlogZ = make_gaussian_ukln_example((0.0, 1.0), (1.0, sigma))
 
-        # estimate 3 times
-        df_ref, ddf_ref = pymbar.BAR(w_F, w_R)
-        df_0, bootstrap_samples = bootstrap_bar(w_F, w_R, n_bootstrap=n_bootstrap)
-        df_1, bootstrap_sigma = bar_with_bootstrapped_uncertainty(w_F, w_R)
+    # estimate 3 times
+    df_ref, df_err_ref = df_and_err_from_u_kln(u_kln)
+    df_0, bootstrap_samples = bootstrap_bar(u_kln, n_bootstrap=n_bootstrap)
+    df_1, bootstrap_sigma = bar_with_bootstrapped_uncertainty(u_kln)
 
-        # assert estimates identical, uncertainties comparable
-        print(f"stddev(w_F) = {sigma_F}, bootstrap uncertainty = {bootstrap_sigma}, pymbar.BAR uncertainty = {ddf_ref}")
-        assert df_0 == df_ref
-        assert df_1 == df_ref
-        assert len(bootstrap_samples) == n_bootstrap, "timed out on default problem size!"
-        np.testing.assert_approx_equal(bootstrap_sigma, ddf_ref, significant=1)
+    # assert estimates identical, uncertainties comparable
+    print(f"bootstrap uncertainty = {bootstrap_sigma}, pymbar.MBAR uncertainty = {df_err_ref}")
+    assert df_0 == df_ref
+    assert df_1 == df_ref
+    assert len(bootstrap_samples) == n_bootstrap, "timed out on default problem size!"
```

Comment:
nit: Is this still relevant?

Reply:
Kept assertion but removed misleading message in https://github.com/proteneer/timemachine/pull/1098/commits/28c6d9ffac52398d2c3f0d0509a1d590c3a1a9d8